[09/15 12:21:13 visual_prompt]: Rank of current process: 0. World size: 1
[09/15 12:21:13 visual_prompt]: Environment info:
-------------------  ----------------------------------------------------
Python               3.7.16 (default, Jan 17 2023, 22:20:44) [GCC 11.2.0]
ENV_MODULE           <not set>
PyTorch              1.7.1
PyTorch Debug Build  False
CUDA available       True
CUDA ID              0
GPU 0                NVIDIA A40
Pillow               9.3.0
cv2                  4.8.0
-------------------  ----------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

[09/15 12:21:13 visual_prompt]: Command line arguments: Namespace(config_file='visual_prompt_tuning/configs/prompt/cub.yaml', opts=['MODEL.TYPE', 'vit', 'DATA.BATCH_SIZE', '64', 'MODEL.PROMPT.NUM_TOKENS', '100', 'MODEL.PROMPT.DEEP', 'True', 'MODEL.PROMPT.DROPOUT', '0.1', 'DATA.FEATURE', 'sup_vitb16_imagenet21k', 'DATA.NAME', 'vtab-resisc45', 'DATA.NUMBER_CLASSES', '45', 'SOLVER.BASE_LR', '5.0', 'SOLVER.WEIGHT_DECAY', '0.0001', 'MODEL.MODEL_ROOT', 'visual_prompt_tuning/model_root', 'DATA.DATAPATH', 'visual_prompt_tuning/data_path', 'OUTPUT_DIR', 'visual_prompt_tuning/output_dir/seed42'], train_type='')
[09/15 12:21:13 visual_prompt]: Contents of args.config_file=visual_prompt_tuning/configs/prompt/cub.yaml:
_BASE_: "../base-prompt.yaml"
RUN_N_TIMES: 1
DATA:
  NAME: "CUB"
  DATAPATH: ""  #TODO: need to specify here
  NUMBER_CLASSES: 200
  MULTILABEL: False
MODEL:
  TYPE: "vit"
SOLVER:
  BASE_LR: 0.1
  WEIGHT_DECAY: 0.01
[09/15 12:21:13 visual_prompt]: Training with config:
[09/15 12:21:13 visual_prompt]: {'CUDNN_BENCHMARK': False,
 'DATA': {'BATCH_SIZE': 64,
          'CLASS_WEIGHTS_TYPE': 'none',
          'CROPSIZE': 224,
          'DATAPATH': 'visual_prompt_tuning/data_path',
          'FEATURE': 'sup_vitb16_imagenet21k',
          'MULTILABEL': False,
          'NAME': 'vtab-resisc45',
          'NO_TEST': False,
          'NUMBER_CLASSES': 45,
          'NUM_WORKERS': 4,
          'PERCENTAGE': 1.0,
          'PIN_MEMORY': True},
 'DBG': False,
 'MODEL': {'ADAPTER': CfgNode({'REDUCATION_FACTOR': 8, 'STYLE': 'Pfeiffer'}),
           'LINEAR': CfgNode({'MLP_SIZES': [], 'DROPOUT': 0.1}),
           'MLP_NUM': 0,
           'MODEL_ROOT': 'visual_prompt_tuning/model_root',
           'PROMPT': {'CLSEMB_FOLDER': '',
                      'CLSEMB_PATH': '',
                      'DEEP': True,
                      'DEEP_SHARED': False,
                      'DROPOUT': 0.1,
                      'FORWARD_DEEP_NOEXPAND': False,
                      'INITIATION': 'random',
                      'LOCATION': 'prepend',
                      'NUM_DEEP_LAYERS': None,
                      'NUM_TOKENS': 100,
                      'PROJECT': -1,
                      'REVERSE_DEEP': False,
                      'SAVE_FOR_EACH_EPOCH': False,
                      'VIT_POOL_TYPE': 'original'},
           'SAVE_CKPT': False,
           'TRANSFER_TYPE': 'prompt',
           'TYPE': 'vit',
           'WEIGHT_PATH': ''},
 'NUM_GPUS': 1,
 'NUM_SHARDS': 1,
 'OUTPUT_DIR': 'visual_prompt_tuning/output_dir/seed42/vtab-resisc45/sup_vitb16_imagenet21k/lr5.0_wd0.0001/run1',
 'RUN_N_TIMES': 1,
 'SEED': None,
 'SOLVER': {'BASE_LR': 5.0,
            'BIAS_MULTIPLIER': 1.0,
            'DBG_TRAINABLE': False,
            'LOG_EVERY_N': 100,
            'LOSS': 'softmax',
            'LOSS_ALPHA': 0.01,
            'MOMENTUM': 0.9,
            'OPTIMIZER': 'sgd',
            'PATIENCE': 300,
            'SCHEDULER': 'cosine',
            'TOTAL_EPOCH': 100,
            'WARMUP_EPOCH': 10,
            'WEIGHT_DECAY': 0.0001,
            'WEIGHT_DECAY_BIAS': 0}}
[09/15 12:21:13 visual_prompt]: Loading training data (final training data for vtab)...
2023-09-15 12:21:13.279710: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-15 12:21:13.472032: I tensorflow/core/util/port.cc:104] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-09-15 12:21:14.446143: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:14.446228: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:14.446237: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-09-15 12:21:16.755606: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:16.755718: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:16.755731: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[09/15 12:21:16 visual_prompt]: Constructing vtab-resisc45 dataset trainval...
2023-09-15 12:21:16.773376: W tensorflow/core/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata".
[INFO: dataset_info.py:  675]: Load pre-computed DatasetInfo (eg: splits, num examples,...) from GCS: resisc45/3.0.0
[INFO: dataset_info.py:  566]: Load dataset info from /tmp/tmpxoepliu_tfds
[INFO: dataset_info.py:  647]: Field info.splits from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.supervised_keys from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.module_name from disk and from code do not match. Keeping the one from code.
[INFO: dataset_builder.py:  575]: Generating dataset resisc45 (visual_prompt_tuning/data_path/resisc45/3.0.0)
Downloading and preparing dataset Unknown size (download: Unknown size, generated: Unknown size, total: Unknown size) to visual_prompt_tuning/data_path/resisc45/3.0.0...
Traceback (most recent call last):
  File "visual_prompt_tuning/train.py", line 129, in <module>
    main(args)
  File "visual_prompt_tuning/train.py", line 124, in main
    train(cfg, args)
  File "visual_prompt_tuning/train.py", line 99, in train
    train_loader, val_loader, test_loader = get_loaders(cfg, logger)
  File "visual_prompt_tuning/train.py", line 66, in get_loaders
    train_loader = data_loader.construct_trainval_loader(cfg)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 79, in construct_trainval_loader
    drop_last=drop_last,
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 31, in _construct_loader
    dataset = TFDataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 78, in __init__
    self.get_data(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 81, in get_data
    tf_data = build_tf_dataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 172, in build_tf_dataset
    vtab_tf_dataloader = data_cls(data_dir=data_dir)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/registry.py", line 50, in __init__
    super(_NewClass, self).__init__(*bound_args, **bound_kwargs)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/resisc45.py", line 46, in __init__
    dataset_builder.download_and_prepare()
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/logging/__init__.py", line 169, in __call__
    return function(*args, **kwargs)
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 630, in download_and_prepare
    download_config=download_config,
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 1437, in _download_and_prepare
    dl_manager, **optional_pipeline_kwargs
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/datasets/resisc45/resisc45_dataset_builder.py", line 98, in _split_generators
    path = os.path.join(dl_manager.manual_dir, 'NWPU-RESISC45')
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/utils/py_utils.py", line 169, in __get__
    cached = self.fget(obj)  # pytype: disable=attribute-error
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/download/download_manager.py", line 704, in manual_dir
    f'Manual directory {self._manual_dir} does not exist or is empty. '
AssertionError: Manual directory visual_prompt_tuning/data_path/downloads/manual does not exist or is empty. Create it and download/extract dataset artifacts in there using instructions:
Dataset can be downloaded from OneDrive:
https://1drv.ms/u/s!AmgKYzARBl5ca3HNaHIlzp_IXjs
After downloading the rar file, please extract it to the manual_dir.
[09/15 12:21:44 visual_prompt]: Rank of current process: 0. World size: 1
[09/15 12:21:44 visual_prompt]: Environment info:
-------------------  ----------------------------------------------------
Python               3.7.16 (default, Jan 17 2023, 22:20:44) [GCC 11.2.0]
ENV_MODULE           <not set>
PyTorch              1.7.1
PyTorch Debug Build  False
CUDA available       True
CUDA ID              0
GPU 0                NVIDIA A40
Pillow               9.3.0
cv2                  4.8.0
-------------------  ----------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

[09/15 12:21:44 visual_prompt]: Command line arguments: Namespace(config_file='visual_prompt_tuning/configs/prompt/cub.yaml', opts=['MODEL.TYPE', 'vit', 'DATA.BATCH_SIZE', '64', 'MODEL.PROMPT.NUM_TOKENS', '100', 'MODEL.PROMPT.DEEP', 'True', 'MODEL.PROMPT.DROPOUT', '0.1', 'DATA.FEATURE', 'sup_vitb16_imagenet21k', 'DATA.NAME', 'vtab-resisc45', 'DATA.NUMBER_CLASSES', '45', 'SOLVER.BASE_LR', '5.0', 'SOLVER.WEIGHT_DECAY', '0.0001', 'MODEL.MODEL_ROOT', 'visual_prompt_tuning/model_root', 'DATA.DATAPATH', 'visual_prompt_tuning/data_path', 'OUTPUT_DIR', 'visual_prompt_tuning/output_dir/seed44'], train_type='')
[09/15 12:21:44 visual_prompt]: Contents of args.config_file=visual_prompt_tuning/configs/prompt/cub.yaml:
_BASE_: "../base-prompt.yaml"
RUN_N_TIMES: 1
DATA:
  NAME: "CUB"
  DATAPATH: ""  #TODO: need to specify here
  NUMBER_CLASSES: 200
  MULTILABEL: False
MODEL:
  TYPE: "vit"
SOLVER:
  BASE_LR: 0.1
  WEIGHT_DECAY: 0.01
[09/15 12:21:44 visual_prompt]: Training with config:
[09/15 12:21:44 visual_prompt]: {'CUDNN_BENCHMARK': False,
 'DATA': {'BATCH_SIZE': 64,
          'CLASS_WEIGHTS_TYPE': 'none',
          'CROPSIZE': 224,
          'DATAPATH': 'visual_prompt_tuning/data_path',
          'FEATURE': 'sup_vitb16_imagenet21k',
          'MULTILABEL': False,
          'NAME': 'vtab-resisc45',
          'NO_TEST': False,
          'NUMBER_CLASSES': 45,
          'NUM_WORKERS': 4,
          'PERCENTAGE': 1.0,
          'PIN_MEMORY': True},
 'DBG': False,
 'MODEL': {'ADAPTER': CfgNode({'REDUCATION_FACTOR': 8, 'STYLE': 'Pfeiffer'}),
           'LINEAR': CfgNode({'MLP_SIZES': [], 'DROPOUT': 0.1}),
           'MLP_NUM': 0,
           'MODEL_ROOT': 'visual_prompt_tuning/model_root',
           'PROMPT': {'CLSEMB_FOLDER': '',
                      'CLSEMB_PATH': '',
                      'DEEP': True,
                      'DEEP_SHARED': False,
                      'DROPOUT': 0.1,
                      'FORWARD_DEEP_NOEXPAND': False,
                      'INITIATION': 'random',
                      'LOCATION': 'prepend',
                      'NUM_DEEP_LAYERS': None,
                      'NUM_TOKENS': 100,
                      'PROJECT': -1,
                      'REVERSE_DEEP': False,
                      'SAVE_FOR_EACH_EPOCH': False,
                      'VIT_POOL_TYPE': 'original'},
           'SAVE_CKPT': False,
           'TRANSFER_TYPE': 'prompt',
           'TYPE': 'vit',
           'WEIGHT_PATH': ''},
 'NUM_GPUS': 1,
 'NUM_SHARDS': 1,
 'OUTPUT_DIR': 'visual_prompt_tuning/output_dir/seed44/vtab-resisc45/sup_vitb16_imagenet21k/lr5.0_wd0.0001/run1',
 'RUN_N_TIMES': 1,
 'SEED': None,
 'SOLVER': {'BASE_LR': 5.0,
            'BIAS_MULTIPLIER': 1.0,
            'DBG_TRAINABLE': False,
            'LOG_EVERY_N': 100,
            'LOSS': 'softmax',
            'LOSS_ALPHA': 0.01,
            'MOMENTUM': 0.9,
            'OPTIMIZER': 'sgd',
            'PATIENCE': 300,
            'SCHEDULER': 'cosine',
            'TOTAL_EPOCH': 100,
            'WARMUP_EPOCH': 10,
            'WEIGHT_DECAY': 0.0001,
            'WEIGHT_DECAY_BIAS': 0}}
[09/15 12:21:44 visual_prompt]: Loading training data (final training data for vtab)...
2023-09-15 12:21:44.939453: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-15 12:21:45.131482: I tensorflow/core/util/port.cc:104] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-09-15 12:21:45.951517: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:45.951597: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:45.951606: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-09-15 12:21:47.678258: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:47.678385: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:21:47.678406: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[09/15 12:21:47 visual_prompt]: Constructing vtab-resisc45 dataset trainval...
2023-09-15 12:21:47.692303: W tensorflow/core/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata".
[INFO: dataset_info.py:  675]: Load pre-computed DatasetInfo (eg: splits, num examples,...) from GCS: resisc45/3.0.0
[INFO: dataset_info.py:  566]: Load dataset info from /tmp/tmpa0che_6ltfds
[INFO: dataset_info.py:  647]: Field info.splits from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.supervised_keys from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.module_name from disk and from code do not match. Keeping the one from code.
[INFO: dataset_builder.py:  575]: Generating dataset resisc45 (visual_prompt_tuning/data_path/resisc45/3.0.0)
Downloading and preparing dataset Unknown size (download: Unknown size, generated: Unknown size, total: Unknown size) to visual_prompt_tuning/data_path/resisc45/3.0.0...
Traceback (most recent call last):
  File "visual_prompt_tuning/train.py", line 129, in <module>
    main(args)
  File "visual_prompt_tuning/train.py", line 124, in main
    train(cfg, args)
  File "visual_prompt_tuning/train.py", line 99, in train
    train_loader, val_loader, test_loader = get_loaders(cfg, logger)
  File "visual_prompt_tuning/train.py", line 66, in get_loaders
    train_loader = data_loader.construct_trainval_loader(cfg)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 79, in construct_trainval_loader
    drop_last=drop_last,
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 31, in _construct_loader
    dataset = TFDataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 78, in __init__
    self.get_data(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 81, in get_data
    tf_data = build_tf_dataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 172, in build_tf_dataset
    vtab_tf_dataloader = data_cls(data_dir=data_dir)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/registry.py", line 50, in __init__
    super(_NewClass, self).__init__(*bound_args, **bound_kwargs)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/resisc45.py", line 46, in __init__
    dataset_builder.download_and_prepare()
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/logging/__init__.py", line 169, in __call__
    return function(*args, **kwargs)
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 630, in download_and_prepare
    download_config=download_config,
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 1437, in _download_and_prepare
    dl_manager, **optional_pipeline_kwargs
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/datasets/resisc45/resisc45_dataset_builder.py", line 98, in _split_generators
    path = os.path.join(dl_manager.manual_dir, 'NWPU-RESISC45')
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/utils/py_utils.py", line 169, in __get__
    cached = self.fget(obj)  # pytype: disable=attribute-error
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/download/download_manager.py", line 704, in manual_dir
    f'Manual directory {self._manual_dir} does not exist or is empty. '
AssertionError: Manual directory visual_prompt_tuning/data_path/downloads/manual does not exist or is empty. Create it and download/extract dataset artifacts in there using instructions:
Dataset can be downloaded from OneDrive:
https://1drv.ms/u/s!AmgKYzARBl5ca3HNaHIlzp_IXjs
After downloading the rar file, please extract it to the manual_dir.
[09/15 12:22:00 visual_prompt]: Rank of current process: 0. World size: 1
[09/15 12:22:00 visual_prompt]: Environment info:
-------------------  ----------------------------------------------------
Python               3.7.16 (default, Jan 17 2023, 22:20:44) [GCC 11.2.0]
ENV_MODULE           <not set>
PyTorch              1.7.1
PyTorch Debug Build  False
CUDA available       True
CUDA ID              0
GPU 0                NVIDIA A40
Pillow               9.3.0
cv2                  4.8.0
-------------------  ----------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

[09/15 12:22:00 visual_prompt]: Command line arguments: Namespace(config_file='visual_prompt_tuning/configs/prompt/cub.yaml', opts=['MODEL.TYPE', 'vit', 'DATA.BATCH_SIZE', '64', 'MODEL.PROMPT.NUM_TOKENS', '100', 'MODEL.PROMPT.DEEP', 'True', 'MODEL.PROMPT.DROPOUT', '0.1', 'DATA.FEATURE', 'sup_vitb16_imagenet21k', 'DATA.NAME', 'vtab-resisc45', 'DATA.NUMBER_CLASSES', '45', 'SOLVER.BASE_LR', '5.0', 'SOLVER.WEIGHT_DECAY', '0.0001', 'MODEL.MODEL_ROOT', 'visual_prompt_tuning/model_root', 'DATA.DATAPATH', 'visual_prompt_tuning/data_path', 'OUTPUT_DIR', 'visual_prompt_tuning/output_dir/seed82'], train_type='')
[09/15 12:22:00 visual_prompt]: Contents of args.config_file=visual_prompt_tuning/configs/prompt/cub.yaml:
_BASE_: "../base-prompt.yaml"
RUN_N_TIMES: 1
DATA:
  NAME: "CUB"
  DATAPATH: ""  #TODO: need to specify here
  NUMBER_CLASSES: 200
  MULTILABEL: False
MODEL:
  TYPE: "vit"
SOLVER:
  BASE_LR: 0.1
  WEIGHT_DECAY: 0.01
[09/15 12:22:00 visual_prompt]: Training with config:
[09/15 12:22:00 visual_prompt]: {'CUDNN_BENCHMARK': False,
 'DATA': {'BATCH_SIZE': 64,
          'CLASS_WEIGHTS_TYPE': 'none',
          'CROPSIZE': 224,
          'DATAPATH': 'visual_prompt_tuning/data_path',
          'FEATURE': 'sup_vitb16_imagenet21k',
          'MULTILABEL': False,
          'NAME': 'vtab-resisc45',
          'NO_TEST': False,
          'NUMBER_CLASSES': 45,
          'NUM_WORKERS': 4,
          'PERCENTAGE': 1.0,
          'PIN_MEMORY': True},
 'DBG': False,
 'MODEL': {'ADAPTER': CfgNode({'REDUCATION_FACTOR': 8, 'STYLE': 'Pfeiffer'}),
           'LINEAR': CfgNode({'MLP_SIZES': [], 'DROPOUT': 0.1}),
           'MLP_NUM': 0,
           'MODEL_ROOT': 'visual_prompt_tuning/model_root',
           'PROMPT': {'CLSEMB_FOLDER': '',
                      'CLSEMB_PATH': '',
                      'DEEP': True,
                      'DEEP_SHARED': False,
                      'DROPOUT': 0.1,
                      'FORWARD_DEEP_NOEXPAND': False,
                      'INITIATION': 'random',
                      'LOCATION': 'prepend',
                      'NUM_DEEP_LAYERS': None,
                      'NUM_TOKENS': 100,
                      'PROJECT': -1,
                      'REVERSE_DEEP': False,
                      'SAVE_FOR_EACH_EPOCH': False,
                      'VIT_POOL_TYPE': 'original'},
           'SAVE_CKPT': False,
           'TRANSFER_TYPE': 'prompt',
           'TYPE': 'vit',
           'WEIGHT_PATH': ''},
 'NUM_GPUS': 1,
 'NUM_SHARDS': 1,
 'OUTPUT_DIR': 'visual_prompt_tuning/output_dir/seed82/vtab-resisc45/sup_vitb16_imagenet21k/lr5.0_wd0.0001/run1',
 'RUN_N_TIMES': 1,
 'SEED': None,
 'SOLVER': {'BASE_LR': 5.0,
            'BIAS_MULTIPLIER': 1.0,
            'DBG_TRAINABLE': False,
            'LOG_EVERY_N': 100,
            'LOSS': 'softmax',
            'LOSS_ALPHA': 0.01,
            'MOMENTUM': 0.9,
            'OPTIMIZER': 'sgd',
            'PATIENCE': 300,
            'SCHEDULER': 'cosine',
            'TOTAL_EPOCH': 100,
            'WARMUP_EPOCH': 10,
            'WEIGHT_DECAY': 0.0001,
            'WEIGHT_DECAY_BIAS': 0}}
[09/15 12:22:00 visual_prompt]: Loading training data (final training data for vtab)...
2023-09-15 12:22:00.708760: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-15 12:22:00.898794: I tensorflow/core/util/port.cc:104] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-09-15 12:22:01.722913: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:01.722987: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:01.722996: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-09-15 12:22:03.439650: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:03.439772: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:03.439788: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[09/15 12:22:03 visual_prompt]: Constructing vtab-resisc45 dataset trainval...
2023-09-15 12:22:03.453938: W tensorflow/core/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata".
[INFO: dataset_info.py:  675]: Load pre-computed DatasetInfo (eg: splits, num examples,...) from GCS: resisc45/3.0.0
[INFO: dataset_info.py:  566]: Load dataset info from /tmp/tmp_eb85idbtfds
[INFO: dataset_info.py:  647]: Field info.splits from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.supervised_keys from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.module_name from disk and from code do not match. Keeping the one from code.
[INFO: dataset_builder.py:  575]: Generating dataset resisc45 (visual_prompt_tuning/data_path/resisc45/3.0.0)
Downloading and preparing dataset Unknown size (download: Unknown size, generated: Unknown size, total: Unknown size) to visual_prompt_tuning/data_path/resisc45/3.0.0...
Traceback (most recent call last):
  File "visual_prompt_tuning/train.py", line 129, in <module>
    main(args)
  File "visual_prompt_tuning/train.py", line 124, in main
    train(cfg, args)
  File "visual_prompt_tuning/train.py", line 99, in train
    train_loader, val_loader, test_loader = get_loaders(cfg, logger)
  File "visual_prompt_tuning/train.py", line 66, in get_loaders
    train_loader = data_loader.construct_trainval_loader(cfg)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 79, in construct_trainval_loader
    drop_last=drop_last,
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 31, in _construct_loader
    dataset = TFDataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 78, in __init__
    self.get_data(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 81, in get_data
    tf_data = build_tf_dataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 172, in build_tf_dataset
    vtab_tf_dataloader = data_cls(data_dir=data_dir)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/registry.py", line 50, in __init__
    super(_NewClass, self).__init__(*bound_args, **bound_kwargs)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/resisc45.py", line 46, in __init__
    dataset_builder.download_and_prepare()
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/logging/__init__.py", line 169, in __call__
    return function(*args, **kwargs)
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 630, in download_and_prepare
    download_config=download_config,
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 1437, in _download_and_prepare
    dl_manager, **optional_pipeline_kwargs
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/datasets/resisc45/resisc45_dataset_builder.py", line 98, in _split_generators
    path = os.path.join(dl_manager.manual_dir, 'NWPU-RESISC45')
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/utils/py_utils.py", line 169, in __get__
    cached = self.fget(obj)  # pytype: disable=attribute-error
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/download/download_manager.py", line 704, in manual_dir
    f'Manual directory {self._manual_dir} does not exist or is empty. '
AssertionError: Manual directory visual_prompt_tuning/data_path/downloads/manual does not exist or is empty. Create it and download/extract dataset artifacts in there using instructions:
Dataset can be downloaded from OneDrive:
https://1drv.ms/u/s!AmgKYzARBl5ca3HNaHIlzp_IXjs
After downloading the rar file, please extract it to the manual_dir.
[09/15 12:22:32 visual_prompt]: Rank of current process: 0. World size: 1
[09/15 12:22:32 visual_prompt]: Environment info:
-------------------  ----------------------------------------------------
Python               3.7.16 (default, Jan 17 2023, 22:20:44) [GCC 11.2.0]
ENV_MODULE           <not set>
PyTorch              1.7.1
PyTorch Debug Build  False
CUDA available       True
CUDA ID              0
GPU 0                NVIDIA A40
Pillow               9.3.0
cv2                  4.8.0
-------------------  ----------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

[09/15 12:22:32 visual_prompt]: Command line arguments: Namespace(config_file='visual_prompt_tuning/configs/prompt/cub.yaml', opts=['MODEL.TYPE', 'vit', 'DATA.BATCH_SIZE', '64', 'MODEL.PROMPT.NUM_TOKENS', '100', 'MODEL.PROMPT.DEEP', 'True', 'MODEL.PROMPT.DROPOUT', '0.1', 'DATA.FEATURE', 'sup_vitb16_imagenet21k', 'DATA.NAME', 'vtab-resisc45', 'DATA.NUMBER_CLASSES', '45', 'SOLVER.BASE_LR', '5.0', 'SOLVER.WEIGHT_DECAY', '0.0001', 'MODEL.MODEL_ROOT', 'visual_prompt_tuning/model_root', 'DATA.DATAPATH', 'visual_prompt_tuning/data_path', 'OUTPUT_DIR', 'visual_prompt_tuning/output_dir/seed100'], train_type='')
[09/15 12:22:32 visual_prompt]: Contents of args.config_file=visual_prompt_tuning/configs/prompt/cub.yaml:
_BASE_: "../base-prompt.yaml"
RUN_N_TIMES: 1
DATA:
  NAME: "CUB"
  DATAPATH: ""  #TODO: need to specify here
  NUMBER_CLASSES: 200
  MULTILABEL: False
MODEL:
  TYPE: "vit"
SOLVER:
  BASE_LR: 0.1
  WEIGHT_DECAY: 0.01
[09/15 12:22:32 visual_prompt]: Training with config:
[09/15 12:22:32 visual_prompt]: {'CUDNN_BENCHMARK': False,
 'DATA': {'BATCH_SIZE': 64,
          'CLASS_WEIGHTS_TYPE': 'none',
          'CROPSIZE': 224,
          'DATAPATH': 'visual_prompt_tuning/data_path',
          'FEATURE': 'sup_vitb16_imagenet21k',
          'MULTILABEL': False,
          'NAME': 'vtab-resisc45',
          'NO_TEST': False,
          'NUMBER_CLASSES': 45,
          'NUM_WORKERS': 4,
          'PERCENTAGE': 1.0,
          'PIN_MEMORY': True},
 'DBG': False,
 'MODEL': {'ADAPTER': CfgNode({'REDUCATION_FACTOR': 8, 'STYLE': 'Pfeiffer'}),
           'LINEAR': CfgNode({'MLP_SIZES': [], 'DROPOUT': 0.1}),
           'MLP_NUM': 0,
           'MODEL_ROOT': 'visual_prompt_tuning/model_root',
           'PROMPT': {'CLSEMB_FOLDER': '',
                      'CLSEMB_PATH': '',
                      'DEEP': True,
                      'DEEP_SHARED': False,
                      'DROPOUT': 0.1,
                      'FORWARD_DEEP_NOEXPAND': False,
                      'INITIATION': 'random',
                      'LOCATION': 'prepend',
                      'NUM_DEEP_LAYERS': None,
                      'NUM_TOKENS': 100,
                      'PROJECT': -1,
                      'REVERSE_DEEP': False,
                      'SAVE_FOR_EACH_EPOCH': False,
                      'VIT_POOL_TYPE': 'original'},
           'SAVE_CKPT': False,
           'TRANSFER_TYPE': 'prompt',
           'TYPE': 'vit',
           'WEIGHT_PATH': ''},
 'NUM_GPUS': 1,
 'NUM_SHARDS': 1,
 'OUTPUT_DIR': 'visual_prompt_tuning/output_dir/seed100/vtab-resisc45/sup_vitb16_imagenet21k/lr5.0_wd0.0001/run1',
 'RUN_N_TIMES': 1,
 'SEED': None,
 'SOLVER': {'BASE_LR': 5.0,
            'BIAS_MULTIPLIER': 1.0,
            'DBG_TRAINABLE': False,
            'LOG_EVERY_N': 100,
            'LOSS': 'softmax',
            'LOSS_ALPHA': 0.01,
            'MOMENTUM': 0.9,
            'OPTIMIZER': 'sgd',
            'PATIENCE': 300,
            'SCHEDULER': 'cosine',
            'TOTAL_EPOCH': 100,
            'WARMUP_EPOCH': 10,
            'WEIGHT_DECAY': 0.0001,
            'WEIGHT_DECAY_BIAS': 0}}
[09/15 12:22:32 visual_prompt]: Loading training data (final training data for vtab)...
2023-09-15 12:22:32.554509: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-15 12:22:32.748613: I tensorflow/core/util/port.cc:104] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-09-15 12:22:33.619542: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:33.619620: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:33.619635: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-09-15 12:22:35.614726: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:35.614835: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:35.614849: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[09/15 12:22:35 visual_prompt]: Constructing vtab-resisc45 dataset trainval...
2023-09-15 12:22:35.627658: W tensorflow/core/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata".
[INFO: dataset_info.py:  675]: Load pre-computed DatasetInfo (eg: splits, num examples,...) from GCS: resisc45/3.0.0
[INFO: dataset_info.py:  566]: Load dataset info from /tmp/tmpahscg0mxtfds
[INFO: dataset_info.py:  647]: Field info.splits from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.supervised_keys from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.module_name from disk and from code do not match. Keeping the one from code.
[INFO: dataset_builder.py:  575]: Generating dataset resisc45 (visual_prompt_tuning/data_path/resisc45/3.0.0)
Downloading and preparing dataset Unknown size (download: Unknown size, generated: Unknown size, total: Unknown size) to visual_prompt_tuning/data_path/resisc45/3.0.0...
Traceback (most recent call last):
  File "visual_prompt_tuning/train.py", line 129, in <module>
    main(args)
  File "visual_prompt_tuning/train.py", line 124, in main
    train(cfg, args)
  File "visual_prompt_tuning/train.py", line 99, in train
    train_loader, val_loader, test_loader = get_loaders(cfg, logger)
  File "visual_prompt_tuning/train.py", line 66, in get_loaders
    train_loader = data_loader.construct_trainval_loader(cfg)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 79, in construct_trainval_loader
    drop_last=drop_last,
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 31, in _construct_loader
    dataset = TFDataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 78, in __init__
    self.get_data(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 81, in get_data
    tf_data = build_tf_dataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 172, in build_tf_dataset
    vtab_tf_dataloader = data_cls(data_dir=data_dir)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/registry.py", line 50, in __init__
    super(_NewClass, self).__init__(*bound_args, **bound_kwargs)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/resisc45.py", line 46, in __init__
    dataset_builder.download_and_prepare()
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/logging/__init__.py", line 169, in __call__
    return function(*args, **kwargs)
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 630, in download_and_prepare
    download_config=download_config,
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 1437, in _download_and_prepare
    dl_manager, **optional_pipeline_kwargs
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/datasets/resisc45/resisc45_dataset_builder.py", line 98, in _split_generators
    path = os.path.join(dl_manager.manual_dir, 'NWPU-RESISC45')
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/utils/py_utils.py", line 169, in __get__
    cached = self.fget(obj)  # pytype: disable=attribute-error
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/download/download_manager.py", line 704, in manual_dir
    f'Manual directory {self._manual_dir} does not exist or is empty. '
AssertionError: Manual directory visual_prompt_tuning/data_path/downloads/manual does not exist or is empty. Create it and download/extract dataset artifacts in there using instructions:
Dataset can be downloaded from OneDrive:
https://1drv.ms/u/s!AmgKYzARBl5ca3HNaHIlzp_IXjs
After downloading the rar file, please extract it to the manual_dir.
[09/15 12:22:57 visual_prompt]: Rank of current process: 0. World size: 1
[09/15 12:22:57 visual_prompt]: Environment info:
-------------------  ----------------------------------------------------
Python               3.7.16 (default, Jan 17 2023, 22:20:44) [GCC 11.2.0]
ENV_MODULE           <not set>
PyTorch              1.7.1
PyTorch Debug Build  False
CUDA available       True
CUDA ID              0
GPU 0                NVIDIA A40
Pillow               9.3.0
cv2                  4.8.0
-------------------  ----------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

[09/15 12:22:57 visual_prompt]: Command line arguments: Namespace(config_file='visual_prompt_tuning/configs/prompt/cub.yaml', opts=['MODEL.TYPE', 'vit', 'DATA.BATCH_SIZE', '64', 'MODEL.PROMPT.NUM_TOKENS', '100', 'MODEL.PROMPT.DEEP', 'True', 'MODEL.PROMPT.DROPOUT', '0.1', 'DATA.FEATURE', 'sup_vitb16_imagenet21k', 'DATA.NAME', 'vtab-resisc45', 'DATA.NUMBER_CLASSES', '45', 'SOLVER.BASE_LR', '5.0', 'SOLVER.WEIGHT_DECAY', '0.0001', 'MODEL.MODEL_ROOT', 'visual_prompt_tuning/model_root', 'DATA.DATAPATH', 'visual_prompt_tuning/data_path', 'OUTPUT_DIR', 'visual_prompt_tuning/output_dir/seed800'], train_type='')
[09/15 12:22:57 visual_prompt]: Contents of args.config_file=visual_prompt_tuning/configs/prompt/cub.yaml:
_BASE_: "../base-prompt.yaml"
RUN_N_TIMES: 1
DATA:
  NAME: "CUB"
  DATAPATH: ""  #TODO: need to specify here
  NUMBER_CLASSES: 200
  MULTILABEL: False
MODEL:
  TYPE: "vit"
SOLVER:
  BASE_LR: 0.1
  WEIGHT_DECAY: 0.01
[09/15 12:22:57 visual_prompt]: Training with config:
[09/15 12:22:57 visual_prompt]: {'CUDNN_BENCHMARK': False,
 'DATA': {'BATCH_SIZE': 64,
          'CLASS_WEIGHTS_TYPE': 'none',
          'CROPSIZE': 224,
          'DATAPATH': 'visual_prompt_tuning/data_path',
          'FEATURE': 'sup_vitb16_imagenet21k',
          'MULTILABEL': False,
          'NAME': 'vtab-resisc45',
          'NO_TEST': False,
          'NUMBER_CLASSES': 45,
          'NUM_WORKERS': 4,
          'PERCENTAGE': 1.0,
          'PIN_MEMORY': True},
 'DBG': False,
 'MODEL': {'ADAPTER': CfgNode({'REDUCATION_FACTOR': 8, 'STYLE': 'Pfeiffer'}),
           'LINEAR': CfgNode({'MLP_SIZES': [], 'DROPOUT': 0.1}),
           'MLP_NUM': 0,
           'MODEL_ROOT': 'visual_prompt_tuning/model_root',
           'PROMPT': {'CLSEMB_FOLDER': '',
                      'CLSEMB_PATH': '',
                      'DEEP': True,
                      'DEEP_SHARED': False,
                      'DROPOUT': 0.1,
                      'FORWARD_DEEP_NOEXPAND': False,
                      'INITIATION': 'random',
                      'LOCATION': 'prepend',
                      'NUM_DEEP_LAYERS': None,
                      'NUM_TOKENS': 100,
                      'PROJECT': -1,
                      'REVERSE_DEEP': False,
                      'SAVE_FOR_EACH_EPOCH': False,
                      'VIT_POOL_TYPE': 'original'},
           'SAVE_CKPT': False,
           'TRANSFER_TYPE': 'prompt',
           'TYPE': 'vit',
           'WEIGHT_PATH': ''},
 'NUM_GPUS': 1,
 'NUM_SHARDS': 1,
 'OUTPUT_DIR': 'visual_prompt_tuning/output_dir/seed800/vtab-resisc45/sup_vitb16_imagenet21k/lr5.0_wd0.0001/run1',
 'RUN_N_TIMES': 1,
 'SEED': None,
 'SOLVER': {'BASE_LR': 5.0,
            'BIAS_MULTIPLIER': 1.0,
            'DBG_TRAINABLE': False,
            'LOG_EVERY_N': 100,
            'LOSS': 'softmax',
            'LOSS_ALPHA': 0.01,
            'MOMENTUM': 0.9,
            'OPTIMIZER': 'sgd',
            'PATIENCE': 300,
            'SCHEDULER': 'cosine',
            'TOTAL_EPOCH': 100,
            'WARMUP_EPOCH': 10,
            'WEIGHT_DECAY': 0.0001,
            'WEIGHT_DECAY_BIAS': 0}}
[09/15 12:22:57 visual_prompt]: Loading training data (final training data for vtab)...
2023-09-15 12:22:57.536847: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-15 12:22:57.740602: I tensorflow/core/util/port.cc:104] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-09-15 12:22:58.628204: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:58.628282: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:22:58.628291: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-09-15 12:23:00.532849: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:23:00.532969: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/cv2/../../lib64:
2023-09-15 12:23:00.532986: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[09/15 12:23:00 visual_prompt]: Constructing vtab-resisc45 dataset trainval...
2023-09-15 12:23:00.546544: W tensorflow/core/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata".
[INFO: dataset_info.py:  675]: Load pre-computed DatasetInfo (eg: splits, num examples,...) from GCS: resisc45/3.0.0
[INFO: dataset_info.py:  566]: Load dataset info from /tmp/tmpza8zvz9gtfds
[INFO: dataset_info.py:  647]: Field info.splits from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.supervised_keys from disk and from code do not match. Keeping the one from code.
[INFO: dataset_info.py:  647]: Field info.module_name from disk and from code do not match. Keeping the one from code.
[INFO: dataset_builder.py:  575]: Generating dataset resisc45 (visual_prompt_tuning/data_path/resisc45/3.0.0)
Downloading and preparing dataset Unknown size (download: Unknown size, generated: Unknown size, total: Unknown size) to visual_prompt_tuning/data_path/resisc45/3.0.0...
Traceback (most recent call last):
  File "visual_prompt_tuning/train.py", line 129, in <module>
    main(args)
  File "visual_prompt_tuning/train.py", line 124, in main
    train(cfg, args)
  File "visual_prompt_tuning/train.py", line 99, in train
    train_loader, val_loader, test_loader = get_loaders(cfg, logger)
  File "visual_prompt_tuning/train.py", line 66, in get_loaders
    train_loader = data_loader.construct_trainval_loader(cfg)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 79, in construct_trainval_loader
    drop_last=drop_last,
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/loader.py", line 31, in _construct_loader
    dataset = TFDataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 78, in __init__
    self.get_data(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 81, in get_data
    tf_data = build_tf_dataset(cfg, split)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/datasets/tf_dataset.py", line 172, in build_tf_dataset
    vtab_tf_dataloader = data_cls(data_dir=data_dir)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/registry.py", line 50, in __init__
    super(_NewClass, self).__init__(*bound_args, **bound_kwargs)
  File "/home/s1952889/final-project/implementation/visual_prompt_tuning/src/data/vtab_datasets/resisc45.py", line 46, in __init__
    dataset_builder.download_and_prepare()
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/logging/__init__.py", line 169, in __call__
    return function(*args, **kwargs)
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 630, in download_and_prepare
    download_config=download_config,
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 1437, in _download_and_prepare
    dl_manager, **optional_pipeline_kwargs
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/datasets/resisc45/resisc45_dataset_builder.py", line 98, in _split_generators
    path = os.path.join(dl_manager.manual_dir, 'NWPU-RESISC45')
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/utils/py_utils.py", line 169, in __get__
    cached = self.fget(obj)  # pytype: disable=attribute-error
  File "/home/s1952889/miniconda3/envs/prompt/lib/python3.7/site-packages/tensorflow_datasets/core/download/download_manager.py", line 704, in manual_dir
    f'Manual directory {self._manual_dir} does not exist or is empty. '
AssertionError: Manual directory visual_prompt_tuning/data_path/downloads/manual does not exist or is empty. Create it and download/extract dataset artifacts in there using instructions:
Dataset can be downloaded from OneDrive:
https://1drv.ms/u/s!AmgKYzARBl5ca3HNaHIlzp_IXjs
After downloading the rar file, please extract it to the manual_dir.
